### PR TITLE
Feature/captions

### PR DESCRIPTION
**Pull request by jponc**
## Changes
- Downloads the file in chunks, built by @KieranP 
- Added captions associations for `Yt::Video` object

### CODE REVIEW

**Comment by yanshiyason** [comment]
I was also in the process of implementing the same feature!

**Comment by jponc** [comment]
@yanshiyason did you manage to implement captions insert?

**Comment by yanshiyason** [comment]
@jponc No, I was trying to retrieve captions from public videos but after
googling a bit it seems like you can only download captions if you are the
owner of the video.. so I was going to do further research on how to bypass
this restriction

**Comment by jponc** [comment]
@yanshiyason When you say `retrieve` Do you want to get a list of captions? Because this PR is built for that.

I read the official Youtube API V3 documentation and they allow caption downloads as long as you provide them the identifier.

https://developers.google.com/youtube/v3/docs/captions/download

**Comment by yanshiyason** [comment]
@jponc the V3 api’s captions endpoint has a ˋlist` method which shows you
metadata about the captions but you must use the ˋdownload` method to see
the actual content. This requires ownership of the video though. This is
what I want to bypass.

**Comment by lucascaton** [comment]
For those who want a workaround to get subtitles quickly (without even having an API key), you can use [youtube-dl](https://github.com/rg3/youtube-dl) with some flags:

```sh
youtube-dl --write-auto-sub --convert-subs=lrc --sub-lang=pt -- <video_id>;
```

Ps.: Replace `--sub-lang=pt` (Portuguese) with `--sub-lang=en` for English or with any other language you want :)

**Comment by yanshiyason** [comment]
Sweet!

**Comment by jponc** [comment]
I can't seem to find an easy way to integrate caption uploads to this PR. Primarily because the "upload" functionality for a video is tightly coupled with thumbnails. Any suggestions?

**Comment by dgb** [comment]
Hi there,

My name is Derek and I'm now one of the maintainers of the Yt gem. I want to both thank you for your contribution and apologize for the lack of feedback on your pull request. Maintaining an open source project like Yt is a pretty big job, and we're taking steps to prioritize maintenance and development of the project. We're hoping to review community contributions on a regular basis, as well as provide a roadmap for our development plans.

As part of this process we're going through the open PRs and taking action on each one, and for this PR we've chosen to review it. You can expect some more specific feedback in the next week or so.

Thanks again for your contribution, and please let us know if you have any questions or comments.

File: lib/yt/models/account.rb
Comment (kangkyu) [inline lib/yt/models/account.rb:86]: it always returns `false` is it intended for development?

File: lib/yt/models/base.rb
Comment (dgb) [inline lib/yt/models/base.rb:28]: I'm not sure I understand if/how this is different from what `open-uri` does.

File: lib/yt/models/account.rb
Comment (dgb) [inline lib/yt/models/account.rb:72]: Is this change necessary? We're using open-uri so this should Just Work ™️ .

File: lib/yt/models/base.rb
Comment (dgb) [inline lib/yt/models/base.rb:1]: This is `require`d elsewhere (in the account model).

**Review by dgb — CHANGES_REQUESTED** [review]
Overall this LGTM but it needs tests before merging. Thanks for bearing with us!

**Comment by jponc** [comment]
Thanks for the recommendations, yeah I haven't really touched the forked branch for quite a while now and I've been really busy lately. Will try next week to do the necessary fix and update this PR. Cheers guys

**Comment by 0xStarcat** [comment]
This feature looks amazing!! :)

**Comment by NicosKaralis** [comment]
Any updates on this?

**Comment by fw42** [comment]
Any chance to get this shipped? Would be awesome. Thanks!

**Comment by jponc** [comment]
Hi guys sorry for the late reply on this, I haven't touched this library (or even Ruby for a while now), so I'm going to close this PR and would just defer the changes to anyone that might be interested.

**Pull request closed without merging**
